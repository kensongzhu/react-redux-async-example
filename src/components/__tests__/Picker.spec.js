import React from "react";
import {shallow} from "enzyme";
import Picker from "../Picker";

const setup = (setupProps = {}) => {

	const defaultProps = {
		value: 'frontend',
		onChange: jest.fn(value => value),
		options: ['frontend', 'react', 'redux']
	};

	const props = {...defaultProps, ...setupProps};

	const wrapper = shallow(<Picker {...props} />);

	return {
		props,
		wrapper
	}
};

describe("Picker", () => {

	test("should render without crashing", () => {
		const {wrapper} = setup();
		expect(wrapper).toMatchSnapshot();
	});

	test("should call onChange callback after select", () => {
		const {props, wrapper} = setup({value: 'foo', options: ['foo', 'bar', 'baz']});
		const h1 = wrapper.find('h1');
		const select = wrapper.find('select');

		// 'foo' as default
		expect(h1.text()).toEqual('foo');
		expect(select.props().value).toEqual('foo');

		// change to `bar`
		// no need to test state, since this component doesn't hold state
		select.simulate('change', {target: {value: 'bar'}});
		expect(props.onChange).toHaveBeenCalledWith('bar');
	})
});