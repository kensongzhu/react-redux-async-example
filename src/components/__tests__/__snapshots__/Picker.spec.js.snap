// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Picker should render without crashing 1`] = `
<span>
  <h1>
    frontend
  </h1>
  <select
    onChange={[Function]}
    value="frontend"
  >
    <option
      key="frontend"
      value="frontend"
    >
      frontend
    </option>
    <option
      key="react"
      value="react"
    >
      react
    </option>
    <option
      key="redux"
      value="redux"
    >
      redux
    </option>
  </select>
</span>
`;
